### PR TITLE
Bug: non squared input images break SklearnClassification task

### PR DESCRIPTION
**Pull request by AlessandroPolidori**
## Summary

Describe the purpose of the pull request, including:

- What problem does it solve?
Non squared input images currently brake sklearn classification task in two ways:
1- torchscript export fails
2- gradcam fails
- How does it solve the problem?
1-?
2- Now mask is reconstructed correctly even when width and height are different.
- Remaining issues
1- I have not tried to solve this problem yet.
2- The mask adjustment is not a trivial task. The reshape() function needs integers, so you need to find two values, height and width, such that w*h = total_size of the mask. But there is no way to know a priori how to truncate height and width values. The solution that satisfied me the most (based on output gradcams) implies a reduction of mask values. I did different tests and the resulting gradcams seem good.

## Type of Change

Please select the one relevant option below:

- Bug fix (non-breaking change that solves an issue)

## Checklist

Please confirm that the following tasks have been completed:

- [x] I have tested my changes locally and they work as expected. (Please describe the tests you performed.)
- [x] I have added unit tests for my changes, or updated existing tests if necessary.
- [x] I have updated the documentation, if applicable.
- [x] I have installed pre-commit and run locally for my code changes.

## Screenshots or Visuals (Optional)

If applicable, please provide screenshots, diagrams, graphs, or videos of the changes, features or the error.

### CODE REVIEW

File: quadra/utils/vit_explainability.py [hunk 1]
```diff
@@ -3,20 +3,24 @@
...
 from typing import List, Optional
 
 import numpy as np
 import torch
 from sklearn.linear_model._base import LinearClassifierMixin
 
 
-def rollout(attentions: List[torch.Tensor], discard_ratio: float = 0.9, head_fusion: str = "mean") -> np.ndarray:
+def rollout(
+    attentions: List[torch.Tensor], discard_ratio: float = 0.9, head_fusion: str = "mean", aspect_ratio: float = 1
```
Comment (lorenzomammana) [inline quadra/utils/vit_explainability.py:15]: If it is float I would put 1.0 as default

File: quadra/utils/vit_explainability.py [hunk 2]
```diff
@@ -120,14 +134,15 @@ def __call__(self, input_tensor: torch.Tensor) -> np.ndarray:
 
 
 def grad_rollout(
-    attentions: List[torch.Tensor], gradients: List[torch.Tensor], discard_ratio: float = 0.9
+    attentions: List[torch.Tensor], gradients: List[torch.Tensor], discard_ratio: float = 0.9, aspect_ratio: float = 1
```
Comment (lorenzomammana) [inline quadra/utils/vit_explainability.py:137]: Same as above

**Review by lorenzomammana — CHANGES_REQUESTED** [review]

**Comment by lorenzomammana** [comment]
https://github.com/pytorch/pytorch/issues/82068 

Related to torchscript fail, we can't do anything about it right now as it is dino code

**Review by lorenzomammana — APPROVED** [review]
